### PR TITLE
Add documentation for special exam events

### PR DESCRIPTION
**Pull request by pdesjardins**
## [DOC-1994](https://openedx.atlassian.net/browse/DOC-1994)

This PR adds documentation for special exam events (proctored, practice proctored, and timed exams).

I added this documentation in a previous PR, which I reverted in favor of waiting for the development team's review of the event behavior. The previous, reverted PR is https://github.com/edx/edx-documentation/pull/805.

Reviewers: After the reverted PR, I made minor changes based on the development team's examination of the event functionality (https://openedx.atlassian.net/browse/PHX-250?focusedCommentId=173667&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-173667). Except for those changes, all of the content is what we reviewed in PR 805. At that time, we had approved the documentation content.
### Reviewers
- [x] Subject matter expert: @douglashall 
- [x] Subject matter expert: @stroilova 
- [ ] Doc team review (sanity check/copy edit/dev edit): @lamagnifica @catong @srpearce 
### Testing
- [ ] Ran ./run_tests.sh without warnings or errors
### HTML Version (optional)
- [x] http://draft-special-exam-events.readthedocs.org/en/latest/
### Post-review
- [ ] Squash commits

### CODE REVIEW

**Comment by douglashall** [comment]
:+1:  Looks good once conflicts are resolved.

File: en_us/data/source/internal_data_formats/special_exam_development_events.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+Practice proctored exams follow a workflow that is similar to proctored exams
+but do not include a review of the exam attempt. Timed exams also follow a
+similar workflow but do not include any proctoring software or review of the
+exam attempt.
+
+The following diagram shows the workflow for a proctored exam.
+
+.. image:: ../../../shared/images/proctored-exam-process.png
+  :width: 500
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_development_events.rst]: is it ok that the image file isn't part of this PR?

File: en_us/data/source/internal_data_formats/special_exam_development_events.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  :alt: A diagram showing the workflow for a proctored exam.
+
+For more information about timed exams and proctored exams, see
+:ref:`partnercoursestaff:Timed Exams` and
+:ref:`partnercoursestaff:CA_ProctoredExams_Overview`.
+
+Some special exam events are emitted when learners interact with exams. For more information, see :ref:`special_exam_events`.
+
+This section includes descriptions of the following events. Some events are
+common to more than one type of special exam.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_development_events.rst]: could researchers misinterpret  "Some events are common to..." as meaning, the same event is emitted by multiple different types of special exams? maybe "Some of the events for the different special exams are (nearly?) identical."

File: en_us/data/source/internal_data_formats/special_exam_development_events.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+.. _special_exam_allowance_deleted:
+
+``edx.special_exam.proctored.allowance.deleted``, ``edx.special_exam.practice.allowance.deleted``, and ``edx.special_exam.timed.allowance.deleted``
+***************************************************************************************************************************************************
+
+The server emits this event when a course team removes an exception to the
+standard exam rules that was previously granted to an individual learner.
+
+The ``event`` fields for this event are the same as the ``event`` fields for
+the ``edx.special_exam.proctored.allowance.created`` event. For more
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_development_events.rst]: maybe replace "proctored" with "{special exam type}"?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+The following diagram shows the workflow for a proctored exam.
+
+.. image:: ../../../shared/images/proctored-exam-process.png
+  :width: 500
+  :alt: A diagram showing the workflow for a proctored exam.
+
+For more information about timed exams and proctored exams, see
+:ref:`partnercoursestaff:Timed Exams` and
+:ref:`partnercoursestaff:CA_ProctoredExams_Overview`.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: hmm. Maybe include the descriptive background (lines 7-41) in just one of the two files (I'd suggest this one), and just xref to here from the other file? or maybe you could set up a subsection rst, and then "include" it into both of these files? An example is "subsection_content_experiments_group_modify_guidelines.rst" which is inserts content into other  files in the olx, course_authors, and open_edx_course_authors guides.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+Some special exam events are emitted when course teams develop and administer exams. For more information, see :ref:`special_exam_development_events`.
+
+This section includes descriptions of the following events. Some events are
+common to more than one type of special exam.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: ditto on the "common" phrasing

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+.. _special_exam_attempt_deleted:
+
+``edx.special_exam.proctored.attempt.deleted``, ``edx.special_exam.practice.attempt.deleted``, and ``edx.special_exam.timed.attempt.deleted``
+*********************************************************************************************************************************************
+
+The server emits this event when a course team or edX platform administrator
+removes an exam attempt record for an individual learner.
+
+The ``event`` fields for this event are the same as the ``event`` fields for
+the ``edx.special_exam.proctored.attempt.created`` event. For more information,
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: I don't want to add micro cleaning, but maybe "{special exam type}" could replace "proctored" for the event descriptions that apply to two or three different events? up to you, I won't continue to comment!

File: en_us/data/source/conf.py [hunk 7]
```diff
@@ -12,3 +12,9 @@
 html_favicon = '../../_themes/edx_theme/static/css/favicon.ico'
 
 project = u'EdX Research Guide'
+
+# Do not attempt to publish .rst files if they are included in
+# other .rst files. This suppresses WARNINGs about documents not
+# being included in any toctree.
+exclude_patterns = ['internal_data_formats/special_exam_events.rst',
+'internal_data_formats/special_exam_development_events.rst']
```
Comment (lamagnifica) [inline en_us/data/source/conf.py]: ah, I was wondering how the files got included!

File: en_us/data/source/internal_data_formats/special_exam_development_events.rst [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+.. _special_exam_updated:
+
+``edx.special_exam.proctored.updated``, ``edx.special_exam.practice.updated``, and ``edx.special_exam.timed.updated``
+*********************************************************************************************************************
+
+The server emits this event when a course team alters a proctored exam in
+Studio.
+
+The ``event`` fields for this event are the same as the ``event`` fields for
+the ``edx.special_exam.proctored.created`` event. For more information, see
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_development_events.rst]: if you choose to change "proctored" above, here too

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+       attempts.
+   * - ``exam_is_practice_exam``
+     - Boolean
+     - Indicates whether the special exam is a practice proctored exam.
+   * - ``exam_is_proctored``
+     - Boolean
+     - Indicates whether the special exam is a proctored exam.
+   * - ``review_attempt_code``
+     - string
+     - A generated identifer for the review of the exam attempt.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: sp identifier

File: en_us/shared/images/proctored-exam-process.svg [hunk 10]
```diff
@@ -0,0 +1,687 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (lamagnifica) [inline en_us/shared/images/proctored-exam-process.svg:1]: ? I'm curious about the SVG, I don't think we've included them in the past. For better rendering? future editing? I would LOVE to have some shared inkscape templates for flowcharting...

**Comment by lamagnifica** [comment]
I think I found a single typo, otherwise a few suggestions and :+1:

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 11]
```diff
@@ -0,0 +1,361 @@
...
+     - The title of the exam in the course page, taken from the subsection
+       title.
+   * - ``exam_is_active``
+     - Boolean
+     - Indicates whether the special exam is currently available for learner
+       attempts.
+   * - ``exam_is_practice_exam``
+     - Boolean
+     - Indicates whether the special exam is a practice proctored exam.
+   * - ``exam_is_proctored``
```
Comment (stroilova) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:262]: Thanks for tagging me @pdesjardins . In general the docs look good -- this is a complicated feature!

Non-blocking question:  
From a design perspective, where did we land here? 
@douglashall  was your team in favor of  e.g. leaving the exam type indicator in both these booleans and in the event name itself? Usually we've been suggesting removing redundant info and considering moving indicators to event payload fields (e.g. event_type = 'proctored'); however, it is possible updates like this are too expensive at this point.

Comment (douglashall) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:262]: @stroilova Yes, I think such an update would be too expensive at this point. We will keep this in mind for future work.

File: en_us/shared/images/proctored-exam-process.svg (hunk 10 above)
Comment (pdesjardins) [inline en_us/shared/images/proctored-exam-process.svg:1]: @lamagnifica I checked in the SVG source file along with the bitmap for future editing. I think it's worth the 31K file size to be able to fix wording more easily. The SVG is not used by the document directly. I agree that a shared template would be great!

**Pull request merged**